core/src/test/java/org/apache/iceberg/hadoop/TestHadoopCommits.java

```diff
@@ -206,6 +191,89 @@ public void testFailedCommit() throws Exception {
+    // commit func should fail
+    assertThatThrownBy(() -> spyOps.commit(metadataV1, metadataV2))
+            .isInstanceOf(CommitFailedException.class)
+            .hasMessageContaining("hdfs crash!");
+    // commit should not successful.
+    int versionAfter = spyOps.findVersion();
+    assert versionBefore == versionAfter;
+  }
+
+  @Test
+  public void testCommitFailedWhenChangeVersionHint() throws Exception {
+    // create v1-format table,for easy testing
+    Map<String,String> confMap = new HashMap<>();
+    confMap.put(TableProperties.FORMAT_VERSION,"1");
+    String newLocation = tableLocation+"v1";
+    Table v1Table = TABLES.create(SCHEMA, SPEC,confMap,newLocation);
+    // Submit a new file to test the functionality.
+    v1Table.newFastAppend().appendFile(FILE_A).commit();
+    BaseTable baseTable = (BaseTable) v1Table;
+    HadoopTableOperations tableOperations = (HadoopTableOperations) baseTable.operations();
+    HadoopTableOperations spyOps = spy(tableOperations);
+    doThrow(new RuntimeException("hdfs crash!"))
+            .when(spyOps)
+            .writeVersionHint(any());
+    int versionBefore = spyOps.findVersion();
+    TableMetadata metadataV1 = spyOps.current();
+    TableMetadata metadataV2 = metadataV1.upgradeToFormatVersion(2);
+    // commit func should fail
+    assertThatThrownBy(() -> spyOps.commit(metadataV1, metadataV2))
+            .isInstanceOf(CommitFailedException.class)
+            .hasMessageContaining("hdfs crash!");
+    // commit should not successful.
+    int versionAfter = spyOps.findVersion();
+    assert versionAfter == versionBefore;
+  }
+
+  @Test
+  public void testCommitFailedAfterChangeVersionHint() throws Exception {
+    // create v1-format table,for easy testing
+    Map<String,String> confMap = new HashMap<>();
+    confMap.put(TableProperties.FORMAT_VERSION,"1");
+    String newLocation = tableLocation+"v1";
+    Table v1Table = TABLES.create(SCHEMA, SPEC,confMap,newLocation);
+    // Submit a new file to test the functionality.
+    v1Table.newFastAppend().appendFile(FILE_A).commit();
+    BaseTable baseTable = (BaseTable) v1Table;
+    HadoopTableOperations tableOperations = (HadoopTableOperations) baseTable.operations();
+    HadoopTableOperations spyOps = spy(tableOperations);
+    doThrow(new RuntimeException("hdfs crash!"))
+            .when(spyOps)
+            .deleteRemovedMetadataFiles(any(),any());
+    int versionBefore = spyOps.findVersion();
+    TableMetadata metadataV1 = spyOps.current();
+    TableMetadata metadataV2 = metadataV1.upgradeToFormatVersion(2);
+    // commit func should fail
+    assertThatThrownBy(() -> spyOps.commit(metadataV1, metadataV2))
+            .isInstanceOf(CommitStateUnknownException.class)
+            .hasMessageContaining("hdfs crash!");
+    // The commit should actually success
+    int versionAfter = spyOps.findVersion();
+    assert versionAfter != versionBefore;
+    assert versionAfter > versionBefore;
```

Comment:
These asserts are redundant, we also should probably just explicitly check the version.

Reply:
Okay. got it